src/main/scala/io/iohk/ethereum/transactions/PendingTransactionsManager.scala

```diff
@@ -88,8 +89,10 @@ class PendingTransactionsManager(txPoolConfig: TxPoolConfig, peerManager: ActorR
 
     case AddOrOverrideTransaction(newStx) =>
       pendingTransactions.cleanUp()
+      // Only validated tranactions are added this way, it is safe to call get
+      val newStxSender = SignedTransaction.getSender(newStx).get
       val obsoleteTxs = pendingTransactions.asMap().asScala.filter(
-        ptx => ptx._2.stx.senderAddress == newStx.senderAddress && ptx._2.stx.tx.nonce == newStx.tx.nonce
+        ptx => SignedTransaction.getSender(ptx._2.stx).contains(newStxSender) && ptx._2.stx.tx.nonce == newStx.tx.nonce
```

Comment:
maybe consider to make method out of this `SignedTransaction.getSender(ptx._2.stx).contains(newStxSender)` because you're using this combination twice already (`PersonalService`)